src/main/kotlin/com/depromeet/piki/user/service/UserService.kt

```
@@ -162,62 +162,51 @@ class UserService(
+    // 신규 user 영속화 공통 경로 — 닉네임 unique 충돌(uq_users_nickname)은 duplicateNickname 으로 매핑한다.
+    // 존재 검사와 save 사이의 race 로 DB 제약 위반이 떠도 500 이 새지 않게 한다.
+    private fun saveNewUser(
+        nickname: String,
+        profileImage: String,
+        identityType: IdentityType,
+    ): User =
+        try {
+            userRepository.save(
+                User(id = UUID.randomUUID(), nickname = nickname, profileImage = profileImage, identityType = identityType),
+            )
+        } catch (e: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)